exchanges/exchange.go

```diff
@@ -1537,7 +1526,39 @@ func (b *Base) GetKlineRequest(pair currency.Pair, a asset.Item, interval kline.
 		return nil, err
 	}
 
-	return kline.CreateKlineRequest(b.Name, pair, formatted, a, interval, exchangeInterval, start, end)
+	limit, err := b.Features.Enabled.Kline.GetIntervalResultLimit(exchangeInterval)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := kline.CreateKlineRequest(b.Name, pair, formatted, a, interval, exchangeInterval, start, end, limit)
+	if err != nil {
+		return nil, err
+	}
+
+	// NOTE: The checks below makes sure a client is notified that using this
+	// functionality will result in error if the total candles cannot be
+	// theoretically retrieved.
+	if fixedAPICandleLength {
+		tn := time.Now().Truncate(exchangeInterval.Duration()).UTC()
+		count := kline.TotalCandlesPerInterval(req.Start, tn, exchangeInterval)
+
+		if count > limit {
+			boundary := tn.Add(-exchangeInterval.Duration() * time.Duration(limit))
+			errMsg := fmt.Sprintf("candle count exceeded: %d. The endpoint has a set candle limit return of %d candles from the request time until %v. Candle data will be incomplete",
```

Comment:
I know I got you to change this, but I didn't know what was coming in terms of UTC alignment, upscaling candles and that the cli `gethistoriccandles` command uses rangesize instead of a startdate.

So take the following:
```cli
 go run . gethistoriccandles --exchange="zb" --pair="btc-usdt" --asset="spot" --rangesize="928"        
2023/04/12 13:52:43 rpc error: code = Unknown desc = candle count exceeded: 5568. The endpoint has a set candle limit return of 1000 candles from the request time until 2022-10-27 08:00:00. Candle data will be incomplete request will exceed exchange limits, please reduce start-end time window or use GetHistoricCandlesExtended
```
I think we could be returning an error that is more clear to both direct callers of wrapper function `GetHistoricCandles` and the CLI command. I was thinking something like:
`Request exceeds API candle limits for 928 24h0m0s candles between 2020-09-26 10:00:00 AEST-2023-04-12 10:00:00 AEST. Request converts to 5569 4h0m0s candles, exceeding the limit of 1000 4h0m0s candles up to 2022-10-27 23:36:51 AEDT. Please reduce timeframe or use GetHistoricCandlesExtended`

Its quite verbose. Its difficult to really communicate why a request failed when behind the scenes it could be converting candles and then goes above the limit. So definitely open to other suggestions, or to just ignore this comment